File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/stats/MeanMilli.java
package com.freight.stats;

/** mean_milli kernel. */
public final class MeanMilli implements StatKernel {

    @Override
    public String name() {
        return "mean_milli";
    }

    @Override
    public long apply(long[] series) {
        if (series.length == 0) {
            return 0L;
        }
        long total = 0L;
        for (int i = 0; i < series.length; i++) {
            total += series[i];
        }
        return StatSupport.floorDiv(total * 1000L, series.length);
    }
}

File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/statx/mean_milli.go
package statx

// MeanMilli is the mean_milli kernel.
func MeanMilli(series []int64) int64 {
	if len(series) == 0 {
		return 0
	}
	total := int64(0)
	for _, value := range series {
		total += value
	}
	return FloorDiv(total*1000, int64(len(series)))
}
